core/src/test/java/org/mobilitydata/gtfsvalidator/table/AnyTableLoaderTest.java

```diff
@@ -0,0 +1,216 @@
+import org.mobilitydata.gtfsvalidator.annotation.FieldLevelEnum;
+import org.mobilitydata.gtfsvalidator.input.CountryCode;
+import org.mobilitydata.gtfsvalidator.input.CurrentDateTime;
+import org.mobilitydata.gtfsvalidator.notice.*;
+import org.mobilitydata.gtfsvalidator.testgtfs.GtfsStop;
+import org.mobilitydata.gtfsvalidator.testgtfs.GtfsStopTableDescriptor;
+import org.mobilitydata.gtfsvalidator.validator.DefaultValidatorProvider;
+import org.mobilitydata.gtfsvalidator.validator.ValidationContext;
+import org.mobilitydata.gtfsvalidator.validator.ValidatorLoader;
+import org.mobilitydata.gtfsvalidator.validator.ValidatorProvider;
+
+public class AnyTableLoaderTest {
+
+  ValidationContext validationContext;
+
+  @Before
+  public void setup() {
+    validationContext =
+        ValidationContext.builder()
+            .setCountryCode(CountryCode.forStringOrUnknown("CA"))
+            .setCurrentDateTime(new CurrentDateTime(ZonedDateTime.now(ZoneId.systemDefault())))
+            .build();
+  }
+
+  @Test
+  public void invalidInputStream() {
+    var testTableDescriptor = mock(GtfsTableDescriptor.class);
+    when(testTableDescriptor.gtfsFilename()).thenReturn("_not_a_valid_file_");
+    NoticeContainer loaderNotices = new NoticeContainer();
+
+    AnyTableLoader.load(
+        testTableDescriptor,
+        mock(ValidatorProvider.class),
+        retrieveInputStream("_not_a_valid_file_"),
+        loaderNotices);
+
+    assertTrue(loaderNotices.hasValidationErrors());
+
+    assertValidationNotice(loaderNotices, CsvParsingFailedNotice.class);
+    assertEquals(1, loaderNotices.getValidationNotices().size());
+    verify(testTableDescriptor, times(0)).createContainerForHeaderAndEntities(any(), any(), any());
+  }
+
+  @Test
+  public void emptyInputStream() {
+    var testTableDescriptor = mock(GtfsTableDescriptor.class);
+    when(testTableDescriptor.gtfsFilename()).thenReturn("filename");
+    NoticeContainer loaderNotices = new NoticeContainer();
+    InputStream csvInputStream = new ByteArrayInputStream("".getBytes(StandardCharsets.UTF_8));
+    ValidatorProvider validatorProvider = mock(ValidatorProvider.class);
+
+    AnyTableLoader.load(testTableDescriptor, validatorProvider, csvInputStream, loaderNotices);
+
+    assertTrue(loaderNotices.hasValidationErrors());
+    assertEquals(1, loaderNotices.getValidationNotices().size());
+    assertThat(loaderNotices.getValidationNotices())
+        .containsExactly(new EmptyFileNotice("filename"));
+    verify(testTableDescriptor, times(0)).createContainerForHeaderAndEntities(any(), any(), any());
+    verify(validatorProvider, times(0)).createSingleFileValidators(any());
+  }
+
+  @Test
+  public void invalidHeaders() {
+    var testTableDescriptor = mock(GtfsTableDescriptor.class);
+    when(testTableDescriptor.gtfsFilename()).thenReturn("filename");
+    GtfsColumnDescriptor[] columns =
+        new GtfsColumnDescriptor[] {
+          AutoValue_GtfsColumnDescriptor.builder()
```

Comment:
It's usually a bad sign if an `AutoValue_` impl class is used outside of the class marked with `@AutoValue` itself.  Prefer to use `GtfsColumDescriptor.builder()` instead.